api/src/main/java/org/apache/gravitino/job/JobTemplate.java

```diff
@@ -0,0 +1,451 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.job;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * JobTemplate is a class to define all the configuration parameters for a job.
+ *
+ * <p>JobType is enum to define the type of the job, because Gravitino needs different runtime
+ * environments to execute different types of jobs, so the job type is required.
+ *
+ * <p>Some parameters can be templated, which means that they can be replaced with actual values
+ * when running the job, for example, arguments can be { "{input_path}}", "{{output_path}" },
+ * environment variables can be { "foo": "{{foo_value}}", "bar": "{{bar_value}}" }. the parameters
+ * support templating are:
+ *
+ * <ul>
+ *   <li>arguments
+ *   <li>environments
+ *   <li>configs
+ * </ul>
+ *
+ * The artifacts (files, jars, archives) will be uploaded to the Gravitino server and managed in the
+ * staging path when registering the job.
+ */
+public final class JobTemplate {
```

Comment:
Take the Iceberg job for example,  how to pass the Iceberg jar paths to Spark command?